js/src/setup-mc/setup-stepper/saved-setup-stepper.js

```diff
@@ -25,6 +33,49 @@ import stepNameKeyMap from './stepNameKeyMap';
 const SavedSetupStepper = ( { savedStep, onRefetchSavedStep = () => {} } ) => {
 	const [ step, setStep ] = useState( savedStep );
 
+	const { settings } = useSettings();
+	const { data: suggestedAudience } = useTargetAudienceWithSuggestions();
+	const {
+		targetAudience,
+		getFinalCountries,
+	} = useTargetAudienceFinalCountryCodes();
+	const {
+		hasFinishedResolution: hasResolvedShippingRates,
+		data: shippingRates,
+	} = useShippingRates();
+	const {
+		hasFinishedResolution: hasResolvedShippingTimes,
+		data: shippingTimes,
+	} = useShippingTimes();
+
+	const { saveTargetAudience, saveSettings } = useAppDispatch();
+	const { saveShippingRates } = useSaveShippingRates();
+	const { saveShippingTimes } = useSaveShippingTimes();
+	const { createNotice } = useDispatchCoreNotices();
+
+	// Auto-save the suggested audience data as the initial values to fall back with the original implementation.
+	// Ref: https://github.com/woocommerce/google-listings-and-ads/blob/2.0.2/js/src/setup-mc/setup-stepper/choose-audience/form-content.js#L37
+	useEffect( () => {
+		if (
+			targetAudience?.location === null &&
+			suggestedAudience?.location
+		) {
+			saveTargetAudience( suggestedAudience );
+		}
+	}, [ targetAudience, suggestedAudience, saveTargetAudience ] );
+
+	// Auto-save the default values for shipping options to fall back with the original implementation.
+	// Ref: https://github.com/woocommerce/google-listings-and-ads/blob/2.0.2/js/src/setup-mc/setup-stepper/setup-free-listings/form-content.js#L33
+	useEffect( () => {
+		if ( settings?.shipping_rate === null ) {
+			saveSettings( {
+				...settings,
+				shipping_rate: 'automatic',
+				shipping_time: 'flat',
```

Comment:
❓ Not a suggestion, just curious. Why we are not adding the default values inside data store DEFAULT_STATE ?

Reply:
I saw we use null values as checkers for other stuff... probably because of that

Reply:
The intent here is to trigger API to save the form's default values. It would be easier and more suitable to handle the intent here.

In addition, moving the default values of form layer to wp-data store layer will also bring different abstraction of data/states isolation. Since `isResolving` and `hasFinishedResolution` are not enough to cover all statuses of a state in wp-store (see issue #438.), and considering the "statuses" of states in wp-data haven't yet had a consistent definition/convention, for now, the default value in a wp-data state is more like the status that tells the state is not ready to be used. And after receiving the data from API, the state will be changed to that data, which will also change the status of that state to "the status of the state is ready, and the data is the state itself".

IMO, before a consistent definition/convention of the statuses, states, and data in wp-data is created, it's probably hard to say how to set a default value would be a good practice.